### PR TITLE
Implement storage service with minio as default provider

### LINKED ISSUE #24: Storage Abstraction Layer - Local FS
**Is your feature request related to a problem? Please describe.**
As the MONAI Deploy Platform utilizes shared storage that could be on a local files system, a mounted volume, a NAS device, or a cloud storage service. MIS needs an abstraction layer defined that allows users to configure the storage that fits their needs or to extend the APIs for other storage services.

**Describe the solution you'd like**
Define an abstraction layer as described in the SRS and implement the default storage access using local FS.  This default option supports local FS, docker mounted volume, and any volume accessible with the local file scheme.

**Describe alternatives you've considered**
Direct access to the storage layer may not be extensible and therefore we are adding an abstraction layer.

### PR DESCRIPTION
**Pull request by mocsharp**
Fixes #24 .

### Description
Introduces new `IStorageService` with default implementation using MinIO. 
Update to the dataflow:
1.  stores all received/downloaded files onto a local temporary storage
2. assembles a new payload object
3. uploads each individual file using the storage service
4. notifies message broker (will be implemented in #25)

Note: a decision was made to save files locally first because,
1. service layer may have high latency, e.g. when uploading to a CSP from on-prem
2. incoming files may be assembled into multiple payloads depending on the grouping rules configured

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [x] All tests passed locally by running `./src/run-tests-in-docker.sh`.
- [x] [Documentation comments](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/documentation-comments) included/updated.
- [x] User guide updated.
- [ ] I have updated the changelog

### CODE REVIEW

File: docs/setup/setup.md [hunk 1]
```diff
@@ -1,54 +1,138 @@
...
+> [!Note]
+> Calculate the required temporary storage based on the number of studies and the size of each study. 
+> Please also consider the AE Title timeout if the AE Title needs to wait a long time before assembling & uploading 
+> the payload for final storage.
+
+
+> [!Note]
+> Before running Informatics Gateway, adjust the values of `watermarkPercent` and `reserveSpaceGB` based on
+> the expected number of studies and size of each study. The suggested value for `reserveSpaceGB` is 2x to 3x the
+> size of a single study multiplied by the number of configured Clara AE Titles.
```
Comment (hshuaib90) [inline docs/setup/setup.md:104]: Clara?

**Review by hshuaib90 — COMMENTED** [review]
Have reviewed the documentation only.

File: docs/setup/setup.md [hunk 2]
```diff
@@ -1,54 +1,138 @@
...
 
 ## Enable Incoming Associations
 
-tbd
+The next step is to configure Informatics Gateway enable receiving of DICOM instances from external DICOM devices.
+
+1. Configure a listening AE Title to receive instances:
+
+```bash
+./mig-cli aet add -a BrainAET -grouping 0020,000E, -t 30
```
[the remark this answers is not in the document]
Comment (mocsharp) [reply docs/setup/setup.md:57]: That's correct.  some institutions may set up a single AE for all incoming associations.  Others may want to skip all the workflow filtering steps by supplying a workflow ID and in such cases, they can name the AET based on the workflow.

File: docs/setup/setup.md (hunk 1 above)
Comment (mocsharp) [reply docs/setup/setup.md]: updated

File: docs/setup/setup.md [hunk 3]
```diff
@@ -1,54 +1,138 @@
...
+
+For complete reference, please refer to the [Config API](../api/rest/config.md).
+
+2. Enable receiving DICOM instances from an external DICOM devices:
+
+```bash
+./mig-cli src add -n PACS-LA -a PACSLA001 --h 20.10.30.55 -p 104
+```
+
+This command above tell Informatics Gateway to accept instances from AT Title `PACSLA001` at IP `20.10.30.55` and port `104`.
```
[the remark this answers is not in the document]
Comment (mocsharp) [reply docs/setup/setup.md]: fixed

File: docs/setup/setup.md [hunk 4]
```diff
@@ -1,54 +1,138 @@
...
+```bash
+./mig-cli aet add -a BrainAET -grouping 0020,000E, -t 30
+```
+
+The command creates a new listening AE Title with AE Title `BrainAET`.  The listening AE Title
+will be grouping instances by the Series Instance UID (0020,000E) with a timeout value of 30 seconds.
+
+For complete reference, please refer to the [Config API](../api/rest/config.md).
+
+2. Enable receiving DICOM instances from an external DICOM devices:
```
[the remark this answers is not in the document]
Comment (mocsharp) [reply docs/setup/setup.md:65]: updated with a note to allow all unknown sources in the same section.

**Review by GreySeaWolf — APPROVED** [review]

**Pull request merged**